Tactics/CSE.lean

```diff
@@ -154,14 +154,24 @@ private def Array.replicate (n : Nat) (v : α) : Array α := List.replicate n v
 def ExprData.isProfitable? (data : ExprData) : CSEM Bool :=
   return data.size > 1 && data.occs >= (← getConfig).minOccsToCSE
 
+/-- Check if an expression is a nat literal, or a `OfNat.ofNat` of a nat literal.
+This lets us avoid CSEing over nat literals, which are already in canonical form.
+-/
+def CSEM.isNatLit (e : Expr) : Bool :=
```

Comment:
Should other literals (e.g., `BitVec`) also deserve the same treatment as `Nat`s here?

Reply:
@shigoel Good point. I'd say morally, yes. I feel that can go in the next PR?

Reply:
Agreed, counting on you to keep track! :)